elftools/elf/segments.py

```diff
@@ -79,6 +79,14 @@ def section_in_segment(self, section):
                 secoffset - poffset + section['sh_size'] <= self['p_filesz'] and
                 secoffset - poffset <= self['p_filesz'] - 1)
 
+    def __str__(self):
+        try: return self.header.p_type
+        except: return super(Segment, self).__str__()
```

Comment:
You don’t want to be handling asynchronous exceptions like KeyboardInterrupt. Change to “except Exception”, or maybe you’re only actually interested in AttributeError or something?